code/object/object.cpp

```diff
@@ -1210,25 +1214,33 @@ void obj_move_all_post(object *objp, float frametime)
 						cast_light = 0;
 					}
 				}
-
-				if ( cast_light )	{
-					weapon_info * wi = &Weapon_info[Weapons[objp->instance].weapon_info_index];
-
-					if ( wi->render_type == WRT_LASER )	{
+				weapon_info* wi = &Weapon_info[Weapons[objp->instance].weapon_info_index];
```

Comment:
Why did you move this outside the if() block?

Reply:
I believe I was at one intermediate stage using something from wi in the if condition, and forgot to moved it back when that became no-longer true. I'll put it back